checkov/terraform/module_loading/loaders/registry_loader.py

```diff
@@ -17,114 +18,101 @@
 )
 from checkov.terraform.module_loading.module_params import ModuleParams
 
+# https://developer.hashicorp.com/terraform/language/modules/sources#fetching-archives-over-http
+MODULE_ARCHIVE_EXTENSIONS = ["zip", "tar.bz2", "tar.gz", "tgz", "tar.xz", "txz"]
+
 
 class RegistryLoader(ModuleLoader):
     modules_versions_cache: Dict[str, List[str]] = {}  # noqa: CCE003  # public data
 
     def __init__(self) -> None:
         super().__init__()
-        self.module_version_url = ""
-        self.best_version = ""
 
     def discover(self, module_params):
-        module_params.REGISTRY_URL_PREFIX = os.getenv("REGISTRY_URL_PREFIX", "https://registry.terraform.io/v1/modules")
-        module_params.token = os.getenv("TFC_TOKEN", "")
+        module_params.tf_host_name = os.getenv("TF_HOST_NAME", TFC_HOST_NAME)
+        module_params.token = os.getenv("TF_REGISTRY_TOKEN", "")
+        tfc_token = os.getenv("TFC_TOKEN")
+        if tfc_token:
+            self.logger.warn("Environment variable TFC_TOKEN will be deprecated in the future. Please use TF_REGISTRY_TOKEN instead.")
+            module_params.token = tfc_token
 
     def _is_matching_loader(self, module_params: ModuleParams) -> bool:
-
-        # Since the registry loader is the first one to be checked,
-        # it shouldn't process any github modules
-        if module_params.module_source.startswith(("github.com", "bitbucket.org", "git::")):
+        if module_params.module_source.startswith("git::"):
```

Comment:
Does this still handle using the special github URLs instead of having to treat it as generic git?

https://developer.hashicorp.com/terraform/language/modules/sources#github

Reply:
This is handled, albeit not very gracefully.  Since github and bitbucket are explicit exceptions, I'll actually revert this line.